src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/cas1/Cas1UpdateSpaceBookingSeedJob.kt

```diff
@@ -0,0 +1,113 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.cas1
+
+import org.slf4j.LoggerFactory
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.ServiceName
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1SpaceBookingEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1SpaceBookingRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CharacteristicRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.SeedColumns
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.SeedJob
+import java.util.UUID
+import kotlin.math.log
+
+@Service
+class Cas1UpdateSpaceBookingSeedJob(
+  val cas1SpaceBookingRepository: Cas1SpaceBookingRepository,
+  val characteristicRepository: CharacteristicRepository,
+) : SeedJob<Cas1UpdateSpaceBookingSeedJobCsvRow>(
+  requiredHeaders = setOf(
+    "space_booking_id",
+    "update_event_number",
+    "event_number",
+    "update_criteria",
+    "criteria",
+  ),
+) {
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  override fun deserializeRow(columns: Map<String, String>): Cas1UpdateSpaceBookingSeedJobCsvRow {
+    val seedColumns = SeedColumns(columns)
+
+    return Cas1UpdateSpaceBookingSeedJobCsvRow(
+      spaceBookingId = seedColumns.uuidOrNull("space_booking_id")!!,
+      updateEventNumber = seedColumns.yesNoBoolean("update_event_number") ?: false,
+      eventNumber = seedColumns.stringOrNull("event_number"),
+      updateCriteria = seedColumns.yesNoBoolean("update_criteria") ?: false,
+      criteria = seedColumns.stringsFromList("criteria"),
+    )
+  }
+
+  override fun processRow(row: Cas1UpdateSpaceBookingSeedJobCsvRow) {
+    val id = row.spaceBookingId
+
+    val spaceBooking = cas1SpaceBookingRepository.findByIdOrNull(id) ?: error("Could not find space booking with id $id")
+
+    if (!row.updateEventNumber && !row.updateCriteria) {
+      error("Nothing to do")
+    }
+
+    updateEventNumber(row, spaceBooking)
+    updateCriteria(row, spaceBooking)
```

Comment:
This could be 
`        if (row.updateEventNumber) updateEventNumber(row, spaceBooking) 
         if (row.updateCriteria) updateSpaceBookingCriteria(row, spaceBooking)
`

Reply:
👍